gamemodes/terrortown/entities/entities/ttt_traitor_button/init.lua

```diff
@@ -0,0 +1,140 @@
+AddCSLuaFile("shared.lua")
+include("shared.lua")
+
+-- serverside only
+ENT.RemoveOnPress = false
+
+ENT.Model = Model("models/weapons/w_bugbait.mdl")
+
+function ENT:Initialize()
+    self:SetModel(self.Model)
+
+    self:SetNoDraw(true)
+    self:DrawShadow(false)
+    self:SetSolid(SOLID_NONE)
+    self:SetMoveType(MOVETYPE_NONE)
+
+    self:SetDelay(self.RawDelay or 1)
+
+    if self:GetDelay() < 0 then
+        -- func_button can be made single use by setting delay to be negative, so
+        -- mimic that here
+        self.RemoveOnPress = true
+    end
+
+    if self.RemoveOnPress then
+        self:SetDelay(-1) -- tells client we're single use
+    end
+
+    if self:GetUsableRange() < 1 then
+        self:SetUsableRange(1024)
+    end
+
+    self:SetNextUseTime(0)
+    self:SetLocked(self:HasSpawnFlags(2048))
+
+    self:SetDescription(self.RawDescription or "?")
+
+    self:SetRole(self.Role or "none")
+    self:SetTeam(self.Team or "none")
+
+    self.RawDelay = nil
+    self.RawDescription = nil
+end
+
+function ENT:KeyValue(key, value)
+    if key == "OnPressed" then
+        self:StoreOutput(key, value)
+    elseif key == "wait" then -- as Delay Before Reset in func_button
+        self.RawDelay = tonumber(value)
+    elseif key == "description" then
+        self.RawDescription = tostring(value)
+
+        if self.RawDescription and string.len(self.RawDescription) < 1 then
+            self.RawDescription = nil
+        end
+    elseif key == "RemoveOnPress" then
+        self[key] = tobool(value)
+    elseif key == "role" then
```

Comment:
Is it right that role is checked at first and - if it is set - the team won't be set?

Reply:
I am pretty sure KeyValue is called once for every KeyValue pair, therefore both will be set.

Reply:
alright